src/oracle/middleware.js

```diff
@@ -0,0 +1,805 @@
+
+      //if interval expired
+      if (timeForPoll){
+        //send a job run with type 1, indicating a job run triggered from the polling interval
+        await submitNewJobIndex(index, 1)
+      }
+    }
+  }, oneSecInterval);
+
+
+  const priceQueryInterval = parseInt(BLOCK_INTERVAL, 10);
+  //validate polling interval
+  assert(!isNaN(priceQueryInterval), `$BLOCK_INTERVAL ${BLOCK_INTERVAL} must be a number`);
+
+  /**
+    * create an interval which query the price and creates a chainlink job request 
+    * if the price deviates more than a specific threshold
+    */
+  const it2 = setInterval(async () => {
+
+    //read state
+    let state = readState();
+
+    //for each job in state
+    for (var i = 0; i < state.jobs.length; i++) {
+      //get the job name
+      let jobName = state.jobs[i].name;
+
+      let sendRequest = 0;
+
+      //query the price
+      let latestPrice = await queryPrice(jobName)
+      let currentPrice = jobName in state.previous_results ? state.previous_results[jobName].result : 0
+
+      //query latest round
+      let latestRound = await queryRound(jobName);
+
+      //reread state in case of updates
+      state = readState();
+
+      //update latest price
+      state.previous_results[jobName].result = latestPrice
+      //update latest round
+      state.previous_results[jobName].round = latestRound
+
+      //if latest round is bigger than last reported round
+      if (latestRound.roundId > state.jobs[getJobIndex(jobName)].last_reported_round) {
+        //if submitted, update last_reported_round
+        if (latestRound.submissionMade) {
+          state.jobs[getJobIndex(jobName)].last_reported_round = latestRound.roundId
+        }
+        //if not found send job request
+        else {
+          console.log("Found new round.")
+          sendRequest = 3
+        }
+      }
+
+      saveJSONDataToFile(state, STATE_FILE)
+
+      //if there's a price deviation
+      let priceDev = Math.abs((latestPrice - currentPrice) / currentPrice) * 100
+
+      if (priceDev > 0){
+        console.log("Found a price deviation for", jobName, "of", priceDev, "%. Latest price:", latestPrice, " Current Price:", currentPrice)
+      }
+
+      //get feed deviation percentage threshold
+      let priceDeviationPercentage = feeds[jobName].priceDeviationPerc
+      if (priceDev > priceDeviationPercentage) {
+        sendRequest = 2
+      }
+
+      //if there is a request to be sent
+      if (sendRequest != 0) {
+
+        //get seconds now
+        let secondsNow = Date.now() / 1000
+        //check seconds passed from last request
+        let secondsPassed = secondsNow - state.jobs[i].last_request_sent
+
+        //check if allowed to send - 12 seconds passed or a request hasnt been made.
+        //12 seconds chosen to wait at least 2 blocks
+        let allowedSend = state.jobs[i].request_id == state.previous_results[jobName].request_id || secondsPassed > Number(SEND_CHECK_INTERVAL)
```

Comment:
Doesn't fit 80 char limit, and type coercion fix
```suggestion
        let allowedSend = state.jobs[i].request_id === state.previous_results[jobName].request_id || secondsPassed > Number(SEND_CHECK_INTERVAL)
```

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/ad4d3cb5cfb0b8e6af0656b9b7e94bf94fdf671c